frontend/components/commons/dataset-description/DatasetDescription.component.vue

```diff
@@ -1,23 +1,65 @@
 <template>
   <div class="description">
     <h2 class="--heading5 --semibold description__title">{{ title }}</h2>
-    <p class="--body1 description__text">{{ datasetDescription }}</p>
+    <BaseSpinner v-if="isLoading" />
+    <MarkdownEditorComponent
+      v-if="!isLoading"
+      :value="datasetGuidelines"
+      placeholder="Insert here your guidelines"
+      @save="updateGuidelines"
+    />
   </div>
 </template>
 
 <script>
+import { getDatasetFromORM } from "@/models/dataset.utilities";
+import { mapActions } from "vuex";
+
 export default {
   props: {
-    datasetDescription: {
+    datasetId: {
+      type: Array,
+      required: true,
+    },
+    datasetTask: {
       type: String,
       required: true,
     },
+    isLoading: {
+      type: Boolean,
+      default: () => true,
+    },
   },
-  data: () => {
+  data() {
     return {
       title: "Description and annotation guidelines",
     };
   },
+  computed: {
+    dataset() {
+      return getDatasetFromORM(this.datasetId, this.datasetTask, false);
+    },
+    datasetGuidelines() {
+      if (this.dataset) {
+        return this.dataset.guidelines
+      }
+      return ''
+    },
+  },
+  methods: {
+    ...mapActions({
+      updateDatasetGuidelines: "entities/datasets/updateDatasetGuidelines",
+    }),
+    async updateGuidelines() {
+      console.log('Saving guidelines')
+      try {
+        await this.updateDatasetGuidelines();
+      }
+      catch (e) {
+        //Do nothing. The updateDatasetGuidelines will fail as there is no backend implementation yet
```

Comment:
Maybe a toast notification can be shown when saving markdown failed. See `Notification` .

Reply:
Added the notification dispatch. Also added a test to verify it works and other improvements in the rest of the tests.